db/migrate/20150819105725_add_lookup_value_match_to_host_and_hostgroup.rb

```diff
@@ -5,10 +5,20 @@ def up
     Host::Managed.reset_column_information
     Hostgroup.reset_column_information
 
-    Host::Managed.where(:type => 'Host::Managed').find_in_batches(:batch_size => 100) do |group|
-      group.each do |host|
-        host.update_attribute(:lookup_value_matcher, host.send(:lookup_value_match))
-      end
+    matchers = LookupValue.where("lookup_values.match like ?", "fqdn=%").group("lookup_values.match").count.keys
+    fqdns = matchers.map{|m| m.gsub(/^fqdn=/, "")}
+    hostnames = fqdns.map{|fqdn| fqdn.split(".").first}
+    if SETTINGS[:unattended]
+      hosts = Host::Managed.where(:nics => {:name => (fqdns + hostnames)}).joins(:primary_interface)
+    else
+      fact_name_id = FactName.where(:name => "fqdn").first.id
+      hosts = Host::Managed.joins(:fact_values).where(:fact_values => {:fact_name_id => fact_name_id, :value => (fqdns + hostnames)})
+    end
+
+    hosts = hosts.readonly(false)
+
+    hosts.uniq.each do |host|
```

Comment:
can probably add the uniqueness to the initial sql query? I would assume its not required here.